clients/githubrepo/graphql.go

```diff
@@ -161,9 +161,9 @@ type checkRunsGraphqlData struct {
 							}
 						} `graphql:"associatedPullRequests(first: $pullRequestsToAnalyze)"`
 					}
-				} `graphql:"history(first: $commitsToAnalyze)"`
+				} `graphql:"history(first:1)"`
```

Comment:
Won't this make the token usage far more expensive? Currently, we retrieve 30 commits in a single graphQL call, this would increase that to 30.

Reply:
From what i have seen in the debugger no, the usage should stay as 1 because the query used for checkRuns, loops over each commit in the handler (After setup has run and populates the commits) and uses ` graphql:"object(oid: $commitExpression)"` to get only one commit. Let me know if this makes sense.

![image](https://user-images.githubusercontent.com/42878263/204569681-3063c498-9251-46b8-a23c-a74bc1e823d5.png)

Reply:
It does result in more individual calls each with a cost of 1, but i have tested with --commit-depth 300 and have had no issues so far.

Reply:
+1 to what @spencerschrock said. We are super conservative with our token usage since that helps us analyze 1M+ repos weekly - https://github.com/ossf/scorecard#public-data

Reply:
oo yeah i didnt think about it regarding the cron. will look for alternatives